js/reference.js

```diff
@@ -5294,18 +5297,22 @@
 
                 // add the main table data to linkedData
                 this._linkedData = [];
+                this._linkedDataRIDs = [];
                 for (i = 0; i < data.length; i++) {
                     tempData = {};
+                    linkedDataMap = {};
                     for (j = 0; j < fks.length; j++) {
                         fkName = fks[j].name;
                         tempData[fkName] = {};
+                        linkedDataMap[fks[j].name] = fks[j].RID
```

Comment:
Please use `fkName` like the line above it. I know these are the same thing, but it might be confusing why we didn't use the same thing later. Also you forgot the `;`.